cleanlab/experimental/datalab/data.py

```diff
@@ -0,0 +1,248 @@
+# Copyright (C) 2017-2023  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+"""Module for class and functions that hold and validate datasets that are loaded into DataLab."""
+
+import os
+from typing import Any, Callable, Dict, List, Mapping, Tuple, Union, cast, TYPE_CHECKING
+
+import datasets
```

Comment:
```suggestion
try:
    import datasets
except ImportError as error:
   raise ImportError("Cannot import datasets package. Please install it and try again, or just install cleanlab with all optional dependencies via: `pip install cleanlab[all]`")
```

Reply:
Let's start a pattern where we do something like this ^ for all optional dependencies

Reply:
once pattern is established, please add guide to DEVELOPMENT.md as well

Reply:
Guide is in DEVELOPMENT.md.

From https://github.com/elisno/cleanlab/commit/33197b3cae8bc60eb7f5ca9b82fcccdfb6c15497.